stream-platform/src/main/java/io/camunda/zeebe/stream/api/records/TypedRecord.java

```diff
@@ -29,4 +31,19 @@ default boolean hasRequestMetadata() {
     return getRequestId() != RecordMetadataEncoder.requestIdNullValue()
         && getRequestStreamId() != RecordMetadataEncoder.requestStreamIdNullValue();
   }
+
+  /**
+   * Returns whether the record was distributed by a different partition. It can do so by decoding
+   * the partition from the key and comparing it against the partition id of this record. If these
+   * match the key was generated by the current partition. If not, the key was generated by a
+   * different partition and then distributed to the current partition. Some commands will not have
+   * a key, indicated by -1. This cannot occur when the command is distributed.
+   *
+   * @return a boolean indicating if the key was generated on the current partition
+   */
+  @JsonIgnore
+  default boolean isCommandDistributed() {
+    final long key = getKey();
+    return key != -1 && Protocol.decodePartitionId(key) != getPartitionId();
+  }
```

Comment:
This method is assuming that it is only called for commands, which breaks [LSP](https://en.wikipedia.org/wiki/Liskov_substitution_principle) because an instance of TypedRecord may be an event or a command_rejection instead of a command. 

❌ We should align the name of this method to the context in which it exists `isDistributed()`, or be more explicit `doesKeyEncodeThisPartitionId()` (don't forget to negate), or make sure to check that the record actually is a command (has my preference).